app/liquid_tags/unified_embed/tag.rb

```diff
@@ -54,12 +54,26 @@ def self.validate_link!(link)
       path = uri.path.presence || "/"
       response = http.request_head(path)
 
-      unless response.is_a?(Net::HTTPSuccess) || response.is_a?(Net::HTTPMovedPermanently)
+      case response
+      when Net::HTTPSuccess
+        response
+      when Net::HTTPRedirection
+        warn "redirected to #{response['location']}"
+      when Net::HTTPNotFound
         raise StandardError, I18n.t("liquid_tags.unified_embed.tag.not_found")
+      else
+        raise StandardError, I18n.t("liquid_tags.unified_embed.tag.invalid_url")
       end
     rescue SocketError
       raise StandardError, I18n.t("liquid_tags.unified_embed.tag.invalid_url")
     end
+
+    def self.extract_only_url(input)
```

Comment:
I decided to create an actual method to process the input.
At first, the only manipulation was to extract the URL if it was passed in with options.
Then I realized that some URLs may have params that could fail the `validate_link!` method.
So I decided to strip those out JUST for the link validation.